README.MD
Vice 2.4.20 LIBRETRO WIP

Very WIP port of VICE 2.4.20 for libretro (using libco).
minimal Libretro arch port (no gui) 

 
Source base: 
http://sourceforge.net/p/vice-emu/code/HEAD/tree/tags/v2.4/v2.4.20/


VICE the Versatile Commodore Emulator

VICE is an emulator collection which emulates the C64, the C64-DTV, the C128, the VIC20, practically all PET models, the PLUS4 and the CBM-II (aka C610). It runs on Unix, MS-DOS, Win32, OS/2, Acorn RISC OS, BeOS, QNX 6.x, Amiga, GP2X or Mac OS X machines.


Developers

Current VICE team members: Dag Lem, Andreas Matthies, Martin Pottendorfer, Marco van den Heuvel, Christian Vogelgsang, Fabrizio Gennari, Daniel Kahlin, Antti S. Lankila, Groepaz, Ingo Korb, Errol Smith, Olaf Seibert, Marcus Sutton, Ulrich Schulz, Stefan Haubenthal, Thomas Giesel, Kajtar Zsolt, Benjamin 'BeRo' Rosseaux.

Of course our warm thanks go to everyone who has helped us in developing VICE during these past few years. For a more detailed list look in the documentation.

Copyright

The VICE is copyrighted to Andreas Boose, Tibor Biczo, Dag Lem, Andreas Dehmel, Andreas Matthies, Martin Pottendorfer, Spiro Trikaliotis, Marco van den Heuvel, Christian Vogelgsang, Fabrizio Gennari, M. Kiesel, Hannu Nuotio, Daniel Kahlin, Antti S. Lankila, Groepaz, Ingo Korb, Errol Smith, Olaf Seibert, Benjamin 'BeRo' Rosseaux, Markus Brenner, David Hansel, Thomas Bretz, Daniel Sladic, Ettore Perazzoli, André Fachat, Teemu Rantanen, Jouko Valta, Jarkko Sonninen.

VICE is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation; either version 2 of the License, or (at your option) any later version.

VICE is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.

--------------------------------------------------------------------------------------------------------------


Use it at you own risk !


Supported platforms : Linux Android

EMUTYPE is limited to : x64 (default) x64sc x64dtv x128

LIBDIR set to /mnt/srdcard/data for Android
and (retro_system_dir)/data for linux
 
Android Build:

make -f Makefile.libretro EMUTYPE=x64sc platform=android




vice-2.4/ChangeLog
------------------------------------------------------------------------------

2015-04-30  Marco van den Heuvel <blackystardust68@yahoo.com>

	* ChangeLog, build/android/build-android.sh, configure.proto,
	doc/vice.chm, doc/vice.guide, doc/vice.hlp, doc/vice.inf,
	doc/vice.pdf, doc/vice.txt, po/da.po, po/de.po, po/es.po, po/fr.po,
	po/hu.po, po/it.po, po/ko.po, po/nl.po, po/pl.po, po/ru.po, po/sv.po,
	po/tr.po, src/arch/android/AnVICE/AndroidManifest.xml,
	src/arch/win32/vice-version.bat, src/version.h: VICE 2.4.20
	

2015-04-30  Marco van den Heuvel <blackystardust68@yahoo.com>

	* po/da.po, po/es.po, po/fr.po, po/hu.po, po/nl.po, po/sv.po,
	src/arch/amigaos/intl_text.c, src/arch/win32/res.rc,
	src/arch/win32/resautostart.rc, src/arch/win32/resc128.rc,
	src/arch/win32/resdqbb.rc, src/arch/win32/resds12c887rtc.rc,
	src/arch/win32/resexpert.rc, src/arch/win32/resgeoram.rc,
	src/arch/win32/reside64.rc, src/arch/win32/resisepic.rc,
	src/arch/win32/resmmc64.rc, src/arch/win32/resmmcreplay.rc,
	src/arch/win32/resmouse.rc, src/arch/win32/resplus4.rc,
	src/arch/win32/resramcart.rc, src/arch/win32/resretroreplay.rc,
	src/arch/win32/resreu.rc, src/arch/win32/ressid.rc,
	src/arch/win32/ressiddtv.rc, src/arch/win32/resuserportrtc.rc,
	src/arch/win32/resvic20.rc, src/translate_text.c: Did a
	'make trans-update'.

2015-04-30  Marco van den Heuvel <blackystardust68@yahoo.com>

	* po/nl.po, po/translations-status.txt: Updated.

2015-04-30  Marco van den Heuvel <blackystardust68@yahoo.com>

	* po/da.po, po/de.po, po/es.po, po/fr.po, po/hu.po, po/it.po,
	po/ko.po, po/nl.po, po/pl.po, po/ru.po, po/sv.po, po/tr.po,
	po/translations-status.txt: Did a 'make update-po'.

2015-04-30  Marco van den Heuvel <blackystardust68@yahoo.com>

	* src/platform/platform_netbsd_runtime_os.c: Updated.

2015-04-29  Marco van den Heuvel <blackystardust68@yahoo.com>

	* src/platform/platform_netbsd_runtime_os.c: Updated.

2015-04-27  groepaz <groepaz@gmx.net>

	* data/C64/Makefile.am, data/C64/win_pos_de.vkm: copied win_pos.vkm to
	win_pos_de.vkm - it really seems to be a map for german layout.
	win_ps.vkm needs to be fixed for us layout.

2015-04-26  groepaz <groepaz@gmx.net>

	* src/lib.c: on error, print a message to stderr instead of silently
	exiting

2015-04-25  groepaz <groepaz@gmx.net>

	* src/arch/win32/archdep.c, src/arch/win32/uikeyboard.c,
	src/keyboard.c, src/log.c: changed logic to select default keymap a
	bit so it will not configure a non existing keymap.

2015-04-24  Marco van den Heuvel <blackystardust68@yahoo.com>

	* src/platform/platform_netbsd_runtime_os.c: Updated.

2015-04-24  Marco van den Heuvel <blackystardust68@yahoo.com>

	* src/platform/platform_linux_runtime_os.c,
	src/platform/platform_netbsd_runtime_os.c: Fixed netbsd detection.

2015-04-23  Marco van den Heuvel <blackystardust68@yahoo.com>

	* src/platform/Makefile.am, src/platform/platform_netbsd_runtime_os.c:
	Added runtime netbsd detection.

2015-04-23  Marco van den Heuvel <blackystardust68@yahoo.com>

	* src/platform/platform_netbsd_version.h: Updated slightly.

2015-04-22  Marco van den Heuvel <blackystardust68@yahoo.com>

	* Makefile.am, config.sub, src/arch/sdl/ui.c, src/opencbm.h,
	src/vice.h, tixbuildinfo: Ported to sortix.

2015-04-20  Marco van den Heuvel <blackystardust68@yahoo.com>

	* src/lib/libffmpeg/libavcodec/x86/ac3dsp_init.c,
	src/lib/libffmpeg/libpostproc/postprocess_template.c: Some (freebsd)
	clang fixes.

2015-04-20  Marco van den Heuvel <blackystardust68@yahoo.com>

	* acinclude.m4, configure.proto, src/lib/ffmpeg/configure.gnu,
	src/lib/libx264/configure: Fixed for when the default compiler is not
	gcc and no gcc is present at all on the system.

2015-04-16  Marcus Sutton <loggedoubt@gmail.com>

	* src/sound.c: Simplified the code to register sound drivers a bit and
	made sure "SoundDeviceName" never gets set to an empty string. It was
	causing the "Default Settings" menu item to use the wrong driver on
	Haiku.

2015-04-16  Marco van den Heuvel <blackystardust68@yahoo.com>

	* src/lib/libffmpeg/libavformat/allformats.c: Fixed.

2015-04-16  Marco van den Heuvel <blackystardust68@yahoo.com>

	* src/platform/platform_x86_gcc_cpuid.h: Made a 'PIC' version of the
	cpuid assembly code.

2015-04-16  Marco van den Heuvel <blackystardust68@yahoo.com>

	* src/opencbm.h: Typedef __u_char for openbsd as well.

2015-04-15  Marco van den Heuvel <blackystardust68@yahoo.com>

	* src/arch/msdos/uidrivec64.c, src/arch/msdos/uidrivec64dtv.c,
	src/arch/msdos/uidriveplus4.c, src/arch/msdos/uidrivevic20.c: Fixed
	some typos.

2015-04-15  Marco van den Heuvel <blackystardust68@yahoo.com>

	* src/arch/msdos/c64ui.c: Fixed a typo.

2015-04-15  Marco van den Heuvel <blackystardust68@yahoo.com>

	* src/c64/cart/ide64.c: Added missing ifdef section.

2015-04-15  groepaz <groepaz@gmx.net>

	* src/arch/sdl/archdep_win32.c: added missing include

2015-04-14  Marco van den Heuvel <blackystardust68@yahoo.com>

	* configure.proto: Fixed for non-resid/c++ build.

2015-04-14  Marco van den Heuvel <blackystardust68@yahoo.com>

	* src/arch/unix/x11/xaw/x11kbd.c: Added missing 'X11/XKBlib.h'
	include.

2015-04-14  Marco van den Heuvel <blackystardust68@yahoo.com>

	* src/arch/sdl/archdep_win32.c: Added guard incase the windows port
	ever gets ported to a windows platform that doesn't support drives.

2015-04-14  groepaz <groepaz@gmx.net>

	* src/arch/sdl/kbd.c: kbd_arch_get_host_mapping should be in archdep
	only

2015-04-13  Marco van den Heuvel <blackystardust68@yahoo.com>

	* src/arch/sdl/archdep_win32.c: Fixed.

2015-04-13  groepaz <groepaz@gmx.net>

	* doc/resource-and-cmdline-checking.txt, doc/vice.pdf, doc/vice.texi,
	doc/vice.txt, src/c128/c128-resources.c: documentation updates related
	to keyboard map rework

2015-04-12  groepaz <groepaz@gmx.net>


libretro/gui/sdlgui.h
/*
	modded for libretro-frodo
*/

/*
  Hatari - sdlgui.h

  This file is distributed under the GNU General Public License, version 2
  or at your option any later version. Read the file gpl.txt for details.

  Header for the tiny graphical user interface for Hatari.
*/
//
#ifndef HATARI_SDLGUI_H
#define HATARI_SDLGUI_H

#include "autostart.h"
#include "vice.h"
#include "lib.h"
#include "machine.h"
#include "ui.h"
#include "attach.h"
#include "util.h"
#include "sid.h"
#include "log.h"
#include "resources.h"
#include "tape.h"
#include "cartridge.h"
#include "imagecontents.h"
#include "tapecontents.h"
#include "diskcontents.h"
#include "videoarch.h"

#define SNAP_BMP 1
#include <strings.h>
#include <string.h>
#include <stdbool.h>
#include <stdio.h>
extern void gui_poll_events(void);

#define SDL_Delay usleep

enum
{
  SGBOX,
  SGTEXT,
  SGEDITFIELD,
  SGBUTTON,
  SGRADIOBUT,
  SGCHECKBOX,
  SGPOPUP,
  SGSCROLLBAR
};


/* Object flags: */
#define SG_TOUCHEXIT   1   /* Exit immediately when mouse button is pressed down */
#define SG_EXIT        2   /* Exit when mouse button has been pressed (and released) */
#define SG_DEFAULT     4   /* Marks a default button, selectable with return key */
#define SG_CANCEL      8   /* Marks a cancel button, selectable with ESC key */

/* Object states: */
#define SG_SELECTED    1
#define SG_MOUSEDOWN   16
#define SG_MOUSEUP     (((int)-1) - SG_MOUSEDOWN)

/* Special characters: */
/*
#define SGRADIOBUTTON_NORMAL    46//0xfa//12
#define SGRADIOBUTTON_SELECTED  219//0xe//13
#define SGCHECKBOX_NORMAL       196//0xf9//14
#define SGCHECKBOX_SELECTED     197// 0x6//15
#define SGARROWUP               203//0x5e// 1
#define SGARROWDOWN             202//0xb//  2
#define SGFOLDER                '~'// 5
*/
#define SGRADIOBUTTON_NORMAL '.'//12
#define SGRADIOBUTTON_SELECTED '*'//13
#define SGCHECKBOX_NORMAL 'o'//14
#define SGCHECKBOX_SELECTED 'X'//15
#define SGARROWUP "^"// 1
#define SGARROWDOWN "v"// 2
#define SGARROWLEFT "<"// 4
#define SGARROWRIGHT ">"// 3
#define SGFOLDER     '~'

/* Return codes: */
#define SDLGUI_ERROR         -1
#define SDLGUI_QUIT          -2
#define SDLGUI_UNKNOWNEVENT  -3


typedef struct
{
  int type;             /* What type of object */
  int flags;            /* Object flags */
  int state;            /* Object state */
  int x, y;             /* The offset to the upper left corner */
  int w, h;             /* Width and height (for scrollbar : height and position) */
  char *txt;            /* Text string */
}  SGOBJ;

extern int sdlgui_fontwidth;	/* Width of the actual font */
extern int sdlgui_fontheight;	/* Height of the actual font */

extern int SDLGui_Init(void);
extern int SDLGui_UnInit(void);
extern int SDLGui_SetScreen();
extern void SDLGui_GetFontSize(int *width, int *height);
extern void SDLGui_Text(int x, int y, const char *txt);
extern void SDLGui_DrawDialog(const SGOBJ *dlg);
extern int SDLGui_DoDialog(SGOBJ *dlg, int *pEventOut);
extern void SDLGui_CenterDlg(SGOBJ *dlg);
extern char* SDLGui_FileSelect(const char *path_and_name, char **zip_path, bool bAllowNew);
extern bool SDLGui_FileConfSelect(char *dlgname, char *confname, int maxlen, bool bAllowNew);

extern void rsxGui_prepare();
extern void rsxGui_flip(int but);

#endif
